src/block/CeilingEdgesHangingSign.php

```diff
@@ -46,6 +47,15 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 			$this->facing = Facing::opposite($player->getHorizontalFacing());
 		}
 
+		$supportBlock = $blockReplace->getSide(Facing::UP);
+		if(
+			$supportBlock->getSupportType(Facing::UP) !== SupportType::FULL &&
+			(!$supportBlock instanceof WallHangingSign || Facing::axis($supportBlock->getFacing()) !== Facing::axis($this->facing)) &&
+			!$supportBlock instanceof CeilingEdgesHangingSign
+		){
+			return false;
+		}
```

Comment:
This should probably be pulled out into proper support logic so the sign self-destructs when these conditions aren't met